projectional/src/main/java/jetbrains/jetpad/projectional/cell/ProjectionalRoleSynchronizer.java

```diff
@@ -34,6 +34,7 @@
   <ContentT> void supportListContentKind(ContentKind<ContentT> kind, Function<ContentT, List<SourceT>> fromContent);
   void setOnLastItemDeleted(Runnable action);
   void setPlaceholderText(String text);
+  void disablePlaceholder(Runnable onLastItemDeleted);
```

Comment:
As I said "onLastItemDeleted should handle the disabled placeholder case". Inside getOnLastItemDeleted() we can check if the placeholder is disabled, we can check if myTargetCellList.getPlaceHolder() is null, etc. 
We already have setOnLastItemDeleted(Runnable action).